crates/graphql/src/schema/query_root.rs

```diff
@@ -694,6 +713,62 @@ impl QueryRoot {
             .collect::<Vec<MetadataJson>>())
     }
 
+    #[graphql(
+        description = "Returns featured collection NFTs ordered by volume (sum of purchase prices)"
+    )]
+    async fn collections_featured_by_volume(
+        &self,
+        context: &AppContext,
+        #[graphql(
+            term = "Return collections whose metadata match this term (case insensitive); sorting occurs among limited search results (rather than searching after sorting)"
+        )]
+        term: Option<String>,
+        #[graphql(order_direction = "Sort ascending or descending")]
+        order_direction: OrderDirection,
+        #[graphql(limit = "Return at most this many results")] limit: i32,
+        #[graphql(offset = "Return results starting from this index")] offset: i32,
+    ) -> FieldResult<Vec<Nft>> {
+        let conn = context.shared.db.get().context("failed to connect to db")?;
+
+        let addresses: Option<Vec<String>> = match term {
+            Some(term) => {
+                let search = &context.shared.search;
+                let search_result = search
+                    .index("collections")
+                    .search()
+                    .with_query(&term)
+                    .with_offset(offset.try_into()?)
+                    .with_limit(limit.try_into()?)
+                    .execute::<Value>()
+                    .await
+                    .context("failed to load search result for collections")?
+                    .hits;
+
+                Some(
+                    search_result
+                        .into_iter()
+                        .map(|r| MetadataJson::from(r.result).mint_address)
+                        .collect(),
+                )
+            },
+            None => None,
+        };
+
+        let collections = queries::collections::by_volume(
+            &conn,
+            addresses,
+            order_direction.into(),
+            limit,
+            offset,
```

Comment:
Probably best to cap the search the results by a fixed number say 100. Or to start no limit since final result chunked by the sql query.

Reply:
@ray-kast Unfortunately there's no way around having unexpected pagination behavior as long as meilisearch happens _before_ the SQL query (and meili results are potentially unbounded). There was a discussion between myself and Mariano in slack to this effect a week or two ago. It's just simply the wrong ordering of operations. See comment above for how I'll try to reduce the impact for now.

Reply:
>I’m happy with how it is currently. For the user best search experience is through meli. It may cut total results on single letter searches like a, f, etc but real unlikely for user to scroll through a 1000 search results.

https://holaplex.slack.com/archives/C0339RA5M3P/p1657043838973919

Discussion in slack.